### ISSUE TITLE
Evaluate Profile-Guided Optimization (PGO)

### ISSUE THREAD

**Issue by zamazan4ik** [groonga/groonga#1585]
### What is your problem?

Not a problem - just an idea of how to  (possibly) improve the performance.

Recently I tested a lot of software with PGO and measured the performance improvements from PGO - the results are [here](https://github.com/zamazan4ik/awesome-pgo/). Since my results show interesting improvements on a lot of applications (including databases) I think it would be a good idea to measure PGO effects on Groonga as well. If the results will show an improvement - would be great to see a note in the documentation about PGO.

**Comment by kou** [comment]
I'm OK with adding build options to enable LTO/PGO/... for easy to try.
But we can't provide the official binaries with PGO enabled because we don't have general workloads for it...

**Comment by zamazan4ik** [comment]
> But we can't provide the official binaries with PGO enabled because we don't have general workloads for it...

Yep. But we can at least test PGO on Groonga. And if it shows performance improvement - write a note about in the Groonga documentation how users can achieve better performance.
